// primevue/listbox/style/listboxstyle.esm.js
import BaseStyle from 'primevue/base/style';

var classes = {
  root: function root(_ref) {
    _ref.instance;
      var props = _ref.props;
    return ['p-listbox p-component', {
      'p-disabled': props.disabled,
      'p-invalid': props.invalid
    }];
  },
  header: 'p-listbox-header',
  filterContainer: 'p-listbox-filter-container',
  filterInput: 'p-listbox-filter p-inputtext p-component',
  filterIcon: 'p-listbox-filter-icon',
  wrapper: 'p-listbox-list-wrapper',
  list: 'p-listbox-list',
  itemGroup: 'p-listbox-item-group',
  item: function item(_ref2) {
    var instance = _ref2.instance,
      option = _ref2.option,
      index = _ref2.index,
      getItemOptions = _ref2.getItemOptions;
    return ['p-listbox-item', {
      'p-highlight': instance.isSelected(option),
      'p-focus': instance.focusedOptionIndex === instance.getOptionIndex(index, getItemOptions),
      'p-disabled': instance.isOptionDisabled(option)
    }];
  },
  emptyMessage: 'p-listbox-empty-message'
};
var ListboxStyle = BaseStyle.extend({
  name: 'listbox',
  classes: classes
});

export { ListboxStyle as default };
